lib/pdf/reader/width_calculator/built_in.rb

```diff
@@ -37,24 +37,13 @@ def initialize(font)
       def glyph_width(code_point)
         return 0 if code_point.nil? || code_point < 0
 
-        m = @metrics.char_metrics_by_code[code_point]
-        if m.nil?
-          names = @font.encoding.int_to_name(code_point)
+        names = @font.encoding.int_to_name(code_point)
+        metrics = names.map { |name|
+          @metrics.char_metrics[name.to_s]
+        }.compact.first
+        return metrics[:wx] if metrics
 
-          m = names.map { |name|
-            @metrics.char_metrics[name.to_s]
-          }.compact.first
-        end
-
-        if m
-          m[:wx]
-        elsif @font.widths[code_point - 1]
-          @font.widths[code_point - 1]
-        elsif control_character?(code_point)
-          0
-        else
-          0
-        end
+        @font.widths[code_point - 1] || 0
```

Comment:
A minor stylistic suggestion based on my own preference to avoid early returns where possible - could this method end like this?

```ruby
if metrics
  metrics[:wx]
else
  @font.widths[code_point - 1] || 0
end
```

Reply:
Sure, I'll change that.